### PR TITLE
4 - Adding localization

### PR DESCRIPTION
**Pull request by irenlian**
# Changes proposed

- Add Intl library for all static text
- Add a `select` component to choose a language
- Add Redux support to store a language
- Add build to CI pipeline

## Demo
Screenshot Before:

Screenshot After:

Screenshot 2020-05-24 at 01 16 05

## New components

N/A

## Caveats

N/A

### CODE REVIEW

File: src/components/Catalog/CatalogPage.tsx [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
+import messages from './messages';
+
+// Redux
+import useAction from '~/hooks/useAction';
 import { getAdsList } from '~/redux/renters/selectors';
+import { getLanguage } from '~/redux/locale/selectors';
 import { addRenterAction } from '~/redux/renters/actions';
-import useAction from '~/hooks/useAction';
+import { changeLanguageAction } from '~/redux/locale/actions';
+import { arrayWithAllLanguages } from '~/redux/locale/types';
```
Comment (dimalian) [inline src/components/Catalog/CatalogPage.tsx:15]: why '~'?

File: src/components/Catalog/CatalogPage.tsx [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+  // changeLanguage action invokes changeLanguage reducer with passed new locale language in payload
+  const changeLanguage = useAction(changeLanguageAction);
 
   return (
-    <div>
-      <h1>Catalog</h1>
-      {adsList.map((ad): JSX.Element => <span>{ad.id}</span>)}
+    <div className="catalog-page">
+      <h1>
+        <FormattedMessage {...messages.title} />
```
Comment (dimalian) [inline src/components/Catalog/CatalogPage.tsx:30]: { ...messages.title }

File: src/components/Catalog/CatalogPage.tsx [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+      </h1>
+      <span>Current language: {usersLocale}</span>
+      <select onChange={(event): void => { changeLanguage(event.target.value); }}>
```
Comment (dimalian) [inline src/components/Catalog/CatalogPage.tsx]: You need to create local Select  which will extend native select and use it.

File: src/components/Catalog/CatalogPage.tsx [hunk 4]
```diff
@@ -1,19 +1,41 @@
...
+        {arrayWithAllLanguages.map(language => (
+          <option value={language} key={language}>{language}</option>
+        ))}
+      </select>
+      <span>Total number of renters ads: {adsList.length}</span>
       <button type="button" onClick={(): void => { addRenter({ id: '1', uuid: '2' }); }}>Add</button>
```
Comment (dimalian) [inline src/components/Catalog/CatalogPage.tsx]: Same like select. Put it to src/common/Button

**Review by dimalian — APPROVED** [review]

File: src/components/Catalog/CatalogPage.tsx (hunk 3 above)
Comment (irenlian) [reply src/components/Catalog/CatalogPage.tsx]: There are just draft components that will be changed in the future.

**Pull request merged**